usr/src/test/zfs-tests/tests/functional/trim/trim.cfg

```diff
@@ -0,0 +1,89 @@
+#!/bin/ksh -p
+#
+#
+# CDDL HEADER START
+#
+# The contents of this file are subject to the terms of the
+# Common Development and Distribution License (the "License").
+# You may not use this file except in compliance with the License.
+#
+# You can obtain a copy of the license at usr/src/OPENSOLARIS.LICENSE
+# or http://www.opensolaris.org/os/licensing.
+# See the License for the specific language governing permissions
+# and limitations under the License.
+#
+# When distributing Covered Code, include this CDDL HEADER in each
+# file and include the License file at usr/src/OPENSOLARIS.LICENSE.
+# If applicable, add the following below this CDDL HEADER, with the
+# fields enclosed by brackets "[]" replaced with your own identifying
+# information: Portions Copyright [yyyy] [name of copyright owner]
+#
+# CDDL HEADER END
+#
+
+#
+# Copyright (c) 2017 by Tim Chase. All rights reserved.
+# Copyright (c) 2017 by Nexenta Systems, Inc. All rights reserved.
+#
+
+#
+# Parameters
+#
+TRIMPOOL="trimpool.$$"
+case "$(uname)" in
+Linux)
+	VDEVDIR="/tmp"
+	VDEVS="/tmp/trim1.dev /tmp/trim2.dev /tmp/trim3.dev /tmp/trim4.dev \
+	    /tmp/trim5.dev"
+
+	HOST_POOL_NAME=''
+	HOST_POOL_DISK=''
+	TRIM_POOL_DISKS="$DISKS"
+	;;
+SunOS)
+	# On Illumos, we can't just shove the files into /tmp, because tmpfs
+	# doesn't support hole punching. UFS doesn't support it either. ZFS
+	# does, but it won't reduce space usage unless the amount of space
+	# freed covers at least a full host FS block (128k in most cases),
+	# which can mess with our space accouting.
+	# To work around these limitations, we simply use the first disk in
+	# $DISKS to hold a host pool with recordsize=4k, so we can guarantee
+	# file hole punching of a usable granularity for our needs.
+	HOST_POOL_NAME="trimhost"
+	HOST_POOL_DISK=$(echo "$DISKS" | awk '{print $1}')
+	TRIM_POOL_DISKS="$(echo "$DISKS" | tr ' ' '\n' | grep -v '^$' | \
+	    tail +2 | tr '\n' ' ')"
+
+	VDEVDIR="/$HOST_POOL_NAME"
+	VDEVS="$VDEVDIR/trim1.dev $VDEVDIR/trim2.dev $VDEVDIR/trim3.dev \
+	    $VDEVDIR/trim4.dev $VDEVDIR/trim5.dev"
+	;;
+esac
+
+# These test limits are algorithm-sensitive, so whenever you adjust the
+# way TRIM processes extents and filters them, be sure to adjust these
+# accordingly to get all tests to pass.
+VDEV_SIZE=256m
+TESTFILE=testfile
+SHRUNK_SIZE_MB=16
```

Comment:
I think 16 is OK but I ended up using 20.  As I'm sure you discovered, the snapshot creation/deletion allocated too much space on higher levels of raidz to allow the smaller values to work.